tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -148,38 +160,10 @@ public void release(T element) {
    * @param element
    */
   public void discard(T element) {
```

Comment:
If there's an element that the calling code determines to be non-reusable, it will call destroy on that element, allowing the pool to create another one if/when necessary.
For instance, if the thrift client has an TException, it discards the thrift client, instead of returning it, since the thrift client may be in a bad state.